src/tests/ftest/cart/test_group_np_cli.c

```diff
@@ -133,11 +133,18 @@ test_run(void)
 
 			for (tag = 0; tag < test_g.t_srv_ctx_num; tag++) {
 				DBG_PRINT("Sending rpc to %d:%d\n", rank, tag);
-				check_in(grp, rank, tag);
+
+				k = test_g.t_num_checkins_to_send;
+				for (j = 0; j < k; j++) {
+					check_in(grp, rank, tag);
+				}
 			}
 		}
 
 		for (i = 0; i < rank_list->rl_nr; i++) {
+			/* Don't wait on RPCs which got aborted. */
```

Comment:
hmm why not? we actually want to ensure that aborted RPCs get their completion callback invoked with a proper error code, it should be -DER_CANCELLED or some such

Reply:
I originally [did assert on `DER_CANCELLED`](https://github.com/daos-stack/daos/pull/4174/commits/183f657eae568baa8fc5f21d70497aee345f8781), but that change was reverted.  I'm not sure if it was reverted due to merging in the swim test changes or another reason, so I'll put that code back where it was.

Reply:
2a75cf73ed80bb5da0e1985fedaaa6d82b450e34